pydm/data_plugins/calc_plugin.py

```diff
@@ -0,0 +1,204 @@
+import functools
+import logging
+import json
+import math
+import numpy as np
+import threading
+
+from pydm.data_plugins.plugin import PyDMPlugin, PyDMConnection
+from qtpy.QtCore import Slot, QThread, Signal, Qt
+from qtpy.QtWidgets import QApplication
+
+import pydm
+
+logger = logging.getLogger(__name__)
+
+
+class CalcThread(QThread):
```

Comment:
:+1: Good this is in a thread. Down the line, for particularly cpu-intensive operations, we could also have a separate plugin that runs these calculations in a separate process via `multiprocessing`.